tools/state-parts-dump-check/src/cli.rs

```diff
@@ -555,7 +565,106 @@ async fn run_single_check(
 
     let duration = start.elapsed();
     tracing::info!("Time elapsed in downloading and validating the parts is: {:?}", duration);
-    Ok(StatePartsDumpCheckStatus::Done { epoch_height: epoch_height })
+    Ok(true)
+}
+
+// download and validate state headers for a single epoch and shard
+async fn check_headers(
+    chain_id: &String,
+    epoch_id: &EpochId,
+    epoch_height: u64,
+    shard_id: ShardId,
+    external: &ExternalConnection,
+) -> anyhow::Result<bool> {
+    let directory_path = external_storage_location_directory(
+        &chain_id,
+        &epoch_id,
+        epoch_height,
+        shard_id,
+        &StateFileType::StateHeader,
+    );
+    tracing::info!(directory_path, "the storage location for the state header being checked:");
+    if !external
+        .is_state_sync_header_stored_for_epoch(shard_id, chain_id, epoch_id, epoch_height)
```

Comment:
this is not a huge deal I guess, but we're calling this function here to see if it exists, and then below in `process_header_with_3_retries` we make a call to the same external storage to retrieve it. Why not just detect that it doesn't exist there? And only retry if the error is retriable. Not a big deal if you don't want to fix it now, since that would require making `ExternalConnection::get_file()` not return an `anyhow:Error`, but instead an error that tells you something about what happened.

Reply:
This is a larger change and we can address it in another PR.